var/spack/repos/builtin/packages/chapel/package.py

```diff
@@ -3,24 +3,813 @@
+        "haswell",
+        "ivybridge",
+        "k8",
+        "k8sse3",
+        "nehalem",
+        "sandybridge",
+        "skylake",
+        "thunderx",
+        "thunderx2t99",
+        "westmere",
+    )
+
+    # TODO: add other package dependencies
+    package_module_dict = {
+        "curl": "curl",
+        "hdf5": "hdf5+hl~mpi",
+        "libevent": "libevent",
+        "protobuf": "py-protobuf",
+        "ssl": "openssl",
+        "yaml": "libyaml@0.1",
+        "zmq": "libzmq",
+    }
+
+    platform_opts = (
+        "cray-cs",
+        "cray-xc",
+        "cygwin32",
+        "cygwin64",
+        "darwin",
+        "hpe-apollo",
+        "hpe-cray-ex",
+        "linux32",
+        "linux64",
+        "netbsd32",
+        "netbsd64",
+        "pwr6",
+        "unset",
+    )
+
+    variant(
+        "atomics",
+        values=("unset", "cstdlib", "intrinsics", "locks"),
+        default="unset",
+        description="Select atomics implementation",
+        multi=False,
+    )
+
+    # TODO: refactor this somehow, this is a separate documentation tool, not a variant of chapel
+    variant("chpldoc", default=False, description="Build chpldoc in addition to chpl")
+
+    variant("developer", default=False, description="Enable Chapel developer mode")
+
+    variant(
+        "comm",
+        default="none",
+        description="Build Chapel with multi-locale support",
+        values=("gasnet", "none", "ofi", "ugni"),
+    )
+
+    variant(
+        "comm_substrate",
+        default="unset",
+        description="Build Chapel with GASNet multi-locale support using the "
+        "supplied CHPL_COMM_SUBSTRATE",
+        values=("ibv", "ofi", "udp", "smp", "unset"),
+        multi=False,
+        sticky=True,  # never allow the concretizer to choose this
+    )
+
+    # Chapel depends on GASNet whenever comm=gasnet.
+    # The default (and recommendation) is to use the embedded copy of GASNet.
+    # This variant allows overriding with a particular version of GASNet sources,
+    # although this is not officially supported and some combinations might be rejected.
+    variant(
+        "gasnet",
+        description="Control the GASNet library version used",
+        default="bundled",
+        values=("bundled", "spack"),
+        multi=False,
+    )
+
+    variant(
+        "gasnet_segment",
+        default="unset",
+        description="Build Chapel with multi-locale support using the "
+        "supplied CHPL_GASNET_SEGMENT",
+        values=("everything", "fast", "large", "unset"),
+        multi=False,
+    )
+
+    variant(
+        "gmp",
+        description="Build with gmp support",
+        default="spack",
+        values=("bundled", "none", "spack"),
+        multi=False,
+    )
+
+    variant(
+        "gpu_mem_strategy",
+        description="The memory allocation strategy for GPU data",
+        values=("array_on_device", "unified_memory"),
+        default="array_on_device",
+        multi=False,
+    )
```

Comment:
(minor) similar theme to other comments: while I see you unsetting the `CHPL_GPU_MEM_STRATEGY` env variable, nothing ever looks at this variant to make a decision, so it is not currently impacting the outcome of the install.